### PR TITLE
Add rule  for no .forward files

### PR DESCRIPTION
**Pull request by teacup-on-rockingchair**
#### Description:

- Add implementation of no_forward_files rule

#### Rationale:

- Add rule to Ensure no users have .forward files together with oval check and  couple tests
- Assign CCE SLE ids for no_forward_files rule
- Add no_forward_files rule to CIS SLE profiles

#### Review Hints:

- No remediation was added, since it might delete sensitive information and also the CIS benchmark seems to recommend it to go without automatic remediation

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  </local_variable>
+
+  <unix:file_object id="object_accounts_users_home_forward_file_existance" version="1">
+    <unix:path var_ref="var_accounts_users_home_forward_file_existance_dirs" var_check="at least one"/>
+    <unix:filename operation="pattern match">\.forward</unix:filename>
+  </unix:file_object>
+
+  <unix:file_test id="test_accounts_users_home_forward_file_existance" check="all"
+                  check_existence="at_least_one_exists" version="1"
+                  comment=".netrc files are not group or world accessible">
```
Comment (Mab879) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml:31]:
```suggestion
                  comment=".forward files are not group or world accessible">
```

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  <local_variable id="var_accounts_users_home_forward_file_existance_dirs" datatype="string" version="1"
+                  comment="Variable including all home dirs from interactive users">
+    <object_component item_field="home_dir"
+                      object_ref="object_accounts_users_home_forward_file_existance_objects"/>
+  </local_variable>
+
+  <unix:file_object id="object_accounts_users_home_forward_file_existance" version="1">
+    <unix:path var_ref="var_accounts_users_home_forward_file_existance_dirs" var_check="at least one"/>
+    <unix:filename operation="pattern match">\.forward</unix:filename>
```
Comment (Mab879) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml:26]:
```suggestion
    <unix:filename operation="pattern match">\.forward$</unix:filename>
```
So we don't have false positives on things like `.forward2` as the benchmark isn't looking for that.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/rule.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+documentation_complete: true
+
```
Comment (Mab879) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/rule.yml:2]: Please add a `prodtype`.

**Review by Mab879 — CHANGES_REQUESTED** [review]

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/rule.yml (hunk 3 above)
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/rule.yml:2]: :+1: added in [478f10d](https://github.com/ComplianceAsCode/content/pull/9990/commits/478f10d91237b340025fe3ca5be521f0347e85d2)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml (hunk 2 above)
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml]: :+1: added in [478f10d](https://github.com/ComplianceAsCode/content/pull/9990/commits/478f10d91237b340025fe3ca5be521f0347e85d2)

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml (hunk 1 above)
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_forward_files/oval/shared.xml]: :man_facepalming: fixed in [478f10d](https://github.com/ComplianceAsCode/content/pull/9990/commits/478f10d91237b340025fe3ca5be521f0347e85d2)

**Review by Mab879 — APPROVED** [review]
LGTM, thanks for the PR

**Comment by Mab879** [comment]
Automatus tests pass locally.

**Pull request merged**